packages/react-hooks/src/useWindow.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright IBM Corp. 2018, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useEventListener } from './useEventListener';
+import { useForceUpdate } from './useForceUpdate';
+import { usePassive } from './usePassive';
+
+/**
+ * Helper hook that will force an update for any window event that occurs. We
+ * force the update so that the calling hook can return values from `window`
+ * directly.
+ *
+ * @param {string} name - the name of the window event
+ * @param {object?} options - optional options for the event listener
+ */
+function useWindowEvent(name, options) {
+  const forceUpdate = useForceUpdate();
+  useEventListener(window, name, forceUpdate, options);
```

Comment:
One technique that might need to be included here is `throttle` support for `forceUpdate`